mantidproject/mantid
# GUI Restructuring 11: Progress bar

~~***PR #25730 should be merged first.***~~

**Description of work.**

Ongoing work to integrate the `JobTreeView` into the `ISIS Reflectometry` GUI. This PR is being merged to a feature branch because not all functionality is implemented yet.

This PR implements the main progress bar. It also includes some consolidation of test code into helper functions. 

The progress bar is largely the same as the original GUI, but there is a small change in that previous progress is now remembered when you pause and restart processing/autoprocessing. Previously when you restarted processing, it would calculate progress only on the remaining rows. This is only applicable when processing the whole table. If you manually select some rows to process the behaviour is the same as before - progress is reported on the current selection.

**To test:**

This should be tested by someone at ISIS.

- Log in to the catalog via the Menu -> catalog login option
- Open the `ISIS Reflectometry` interface
- Set the instrument to `INTER` and enter `1120015` in the search box and click `Autoprocess`
- All runs for the experiment should appear in the main table and processing should begin. The main progress bar should start incrementing.
- Pause autoprocessing and restart it (note that it may take a while to pause as it finishes the current job). The progress bar should continue from where it left off.
- Pause again and this take make sure there are no rows selected in the table and click the `Process` button. Again the progress bar should continue from where it left off.
- Pause again and select some rows/groups and and click `Process`. This time the progress bar should report on progress of the selection only.

Refs #23027

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
